controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -289,6 +289,9 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 			NodeDrainTimeout: kcp.Spec.MachineTemplate.NodeDrainTimeout,
 		},
 	}
+	if kcp.Spec.MachineTemplate.NodeDeletionTimeout != nil {
```

Comment:
The KCP uses [`MatchesMachineSpec()`](https://github.com/kubernetes-sigs/cluster-api/blob/v1.0.2/controlplane/kubeadm/internal/filters.go#L33) to identify if there was a config change, and that only respects metadata, `cluster.x-k8s.io/cloned-from-{name,group}` annotations and infrastructure ref.

That means if you update the deletion timeout you would need to manually trigger a rollout. The same applies to NodeDrainTimeout and FailureDomain.

Reply:
Perfect thx. Same behavior as for the other fields sound reasonable for now.

Reply:
That stuff probably doesn't get changed often, but then it might be even more confusing if it doesn't get applied automatically. Should also be fairly straightforward to change that.